include/libtorrent/file_storage.hpp

```diff
@@ -134,7 +134,7 @@ namespace libtorrent
 		};
 
 		// the offset of this file inside the torrent
-		std::uint64_t offset:48;
+		std::int64_t offset:48;
```

Comment:
it seems clang knows that 48 bits can't overflow a signed of 64, don't complain without the cast, let's see msvc and gcc